include/crow/websocket.h

```diff
@@ -390,7 +390,15 @@ namespace crow
                     }
                     break;
                     case WebSocketReadState::Mask:
-                        if (has_mask_)
+                        if (remaining_length_ > max_payload_bytes_)
+                        {
+                            close_connection_ = true;
+                            adaptor_.close();
```

Comment:
This close() call will remove the ability to call get_remote_ip() in the onclose() call the user has customized.

I'm unsure if we have to close the adapter first. Can we error_handler_() and check_destroy() first then close the adaptor?

Reply:
This is a completely different issue to be completely honest. What I would do is copy the ip address to the websocket connection class as soon as the adaptor is available. That way closing the adaptor won't affect the connection object in any way.

Reply:
The Idea with close and error handlers is that they're meant to do any required cleanup without sending or receiving data.